test/core/CMakeLists.txt

```
@@ -10,3 +10,8 @@ add_test(NAME executor_test COMMAND test_executor)
 add_executable(test_executor "test_Executor.cpp" "../test_main.cpp")
 target_link_libraries(test_executor PRIVATE Catch2::Catch2 NeoFOAM
                                             Kokkos::kokkos)
+
+add_test(NAME register_test COMMAND test_RegisterClass)
+add_executable(test_RegisterClass "registerClass.cpp" "../test_main.cpp")
+target_link_libraries(test_RegisterClass PRIVATE Catch2::Catch2 NeoFOAM
+                                                 Kokkos::kokkos)
```

I added a cmake function to handle unit test cases. if you rebase this PR onto newest main you can replace all of it with 
```suggestion

neofoamunittest(registerClass)